src/rhsmlib/facts/network.py

```diff
@@ -0,0 +1,176 @@
+                        ipv4_broadcasts.append(address["broadcast"])
+                    else:
+                        # FIXME Should localhost's broadcast be simply omitted?
+                        ipv4_broadcasts.append("Unknown")
+                    ipv4_netmasks.append(address["prefixlen"])
+
+                if address["family"] == "inet6":
+                    if address["scope"] == "global":
+                        ipv6_global_addresses.append(address["local"])
+                        ipv6_global_netmasks.append(address["prefixlen"])
+                    if address["scope"] == "link":
+                        ipv6_link_addresses.append(address["local"])
+                        ipv6_link_netmasks.append(address["prefixlen"])
+                    if address["scope"] == "host":
+                        ipv6_host_addresses.append(address["local"])
+                        ipv6_host_netmasks.append(address["prefixlen"])
+
+            if ipv4_addresses:
+                result[f"{prefix}.ipv4_address"] = ipv4_addresses[0]
+                result[f"{prefix}.ipv4_address_list"] = " ".join(ipv4_addresses)
+            if ipv4_broadcasts:
+                result[f"{prefix}.ipv4_broadcast"] = ipv4_broadcasts[0]
+                result[f"{prefix}.ipv4_broadcast_list"] = " ".join(ipv4_broadcasts)
+            if ipv4_netmasks:
+                result[f"{prefix}.ipv4_netmask"] = str(ipv4_netmasks[0])
+                result[f"{prefix}.ipv4_netmask_list"] = " ".join(str(mask) for mask in ipv4_netmasks)
+
+            if ipv6_global_addresses:
+                result[f"{prefix}.ipv6_address.global"] = ipv6_global_addresses[0]
+                result[f"{prefix}.ipv6_address.global_list"] = " ".join(ipv6_global_addresses)
+            if ipv6_link_addresses:
+                result[f"{prefix}.ipv6_address.link"] = ipv6_link_addresses[0]
+                result[f"{prefix}.ipv6_address.link_list"] = " ".join(ipv6_link_addresses)
+            if ipv6_host_addresses:
+                result[f"{prefix}.ipv6_address.host"] = ipv6_host_addresses[0]
+                result[f"{prefix}.ipv6_address.host_list"] = " ".join(ipv6_host_addresses)
+            if ipv6_global_netmasks:
+                result[f"{prefix}.ipv6_netmask.global"] = str(ipv6_global_netmasks[0])
+                result[f"{prefix}.ipv6_netmask.global_list"] = " ".join(
+                    str(mask) for mask in ipv6_global_netmasks
+                )
+            if ipv6_link_netmasks:
+                result[f"{prefix}.ipv6_netmask.link"] = str(ipv6_link_netmasks[0])
+                result[f"{prefix}.ipv6_netmask.link_list"] = " ".join(
+                    str(mask) for mask in ipv6_link_netmasks
+                )
+            if ipv6_host_netmasks:
+                result[f"{prefix}.ipv6_netmask.host"] = str(ipv6_host_netmasks[0])
+                result[f"{prefix}.ipv6_netmask.host_list"] = " ".join(
+                    str(mask) for mask in ipv6_host_netmasks
+                )
```

Comment:
hm what about deduplicating the code for these a bit? ie something like:
```python
def add_fact(name, items):
    if items:
        result[f"{prefix}.{name}"] = str(items[0])
        result[f"{prefix}.{name}_list"] = " ".join(str(i) for i in items)

add_fact("ipv4_address", ipv4_addresses)
```
since the logic is the same for all the facts, this places that logic in a single function, and avoids typing issues (eg a wrong variable, or a wrong fact name)